pkg/apis/runtime/runtime.go

```diff
@@ -0,0 +1,56 @@
+package runtime
+
+import (
+	"github.com/equinor/radix-operator/pkg/apis/defaults"
+	radixv1 "github.com/equinor/radix-operator/pkg/apis/radix/v1"
+)
+
+// NodeTypeDeprecation holds information about deprecation of a node type
+type NodeTypeDeprecation struct {
+	// Message is the deprecation message
+	Message string `json:"message,omitempty"`
+	// ReplacedBy is the node type that replaces this node type
+	ReplacedBy string `json:"replacedBy,omitempty"`
+}
+
+// NodeType holds information about a node type attributes
+type NodeType struct {
+	// Architecture is the architecture of the node type
+	Architecture radixv1.RuntimeArchitecture `json:"architecture,omitempty"`
+	// Deprecated is the deprecation information of the node type
+	Deprecated *NodeTypeDeprecation `json:"deprecated,omitempty"`
+}
+
+const (
+	// NodeTypeAffinityKey is the label key for the custom node type
+	NodeTypeAffinityKey = "radix-nodetype"
+	// NodeTypeTolerationKey is the taint key for the custom node type
+	NodeTypeTolerationKey = "radix-nodetype"
+)
+
+// Node types and their attributes supported by Radix
+var nodeTypes = map[string]NodeType{
+	"memory-optimized-v1": {Architecture: radixv1.RuntimeArchitectureAmd64},
```

Comment:
We talked abit more this morning, and think it would be a good idea to have the same naming convention as in gpus, so to add a `2` for 2Tb of memory:

```suggestion
	"memory-optimized-2-v1": {Architecture: radixv1.RuntimeArchitectureAmd64},
```